File: app/src/main/java/com/friday/ar/util/BootReciever.java
package com.friday.ar.util;

import android.app.job.JobInfo;
import android.app.job.JobScheduler;
import android.content.BroadcastReceiver;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.friday.ar.FridayApplication;
import com.friday.ar.plugin.application.PluginLoader;
import com.friday.ar.service.AccountSyncService;
import com.friday.ar.service.PluginIndexer;

public class BootReciever extends BroadcastReceiver {
    private static final String LOGTAG = "BootReciever";

    @Override
    public void onReceive(final Context context, Intent pIntent) {
        JobScheduler jobScheduler = (JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
        if (preferences.getBoolean("check_update_auto", false)) {
            UpdateUtil.checkForUpdate(context);
        }
        if (preferences.getBoolean("sync_account_auto", true)) {
            JobInfo info = new JobInfo.Builder(FridayApplication.Jobs.JOB_SYNC_ACCOUNT, new ComponentName(context, AccountSyncService.class))
                    .setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY)
                    .setBackoffCriteria(2 * 60000, JobInfo.BACKOFF_POLICY_LINEAR)
                    .build();
            jobScheduler.schedule(info);
        }
        PluginLoader pluginLoader = new PluginLoader(context);
        if (pluginLoader.startLoading()) {

        }

        JobInfo jobIndexerInfo = new JobInfo.Builder(FridayApplication.Jobs.JOB_INDEX_PLUGINS, new ComponentName(context, PluginIndexer.class))
                .setBackoffCriteria(30 * 60000, JobInfo.BACKOFF_POLICY_LINEAR)
                .build();
        jobScheduler.schedule(jobIndexerInfo);
    }
}

File: app/src/main/java/com/friday/ar/util/BootReciever.kt
package com.friday.ar.util

import android.app.job.JobInfo
import android.app.job.JobScheduler
import android.content.*
import android.os.Build
import com.friday.ar.account.sync.AccountSyncService
import com.friday.ar.core.Constant
import com.friday.ar.pluginsystem.service.PluginLoader
import org.koin.core.KoinComponent
import org.koin.core.get

class BootReciever : BroadcastReceiver(), KoinComponent {

    override fun onReceive(context: Context, pIntent: Intent) {
        if (pIntent.action != Intent.ACTION_BOOT_COMPLETED) return
        val jobScheduler = context.getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
        val preferences: SharedPreferences = get()
        if (preferences.getBoolean("check_update_auto", false)) {

        }
        if (preferences.getBoolean("sync_account_auto", true)) {
            val info = JobInfo.Builder(Constant.Jobs.JOB_SYNC_ACCOUNT, ComponentName(context, AccountSyncService::class.java))
                    .setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY)
                    .setBackoffCriteria((2 * 60000).toLong(), JobInfo.BACKOFF_POLICY_LINEAR)
                    .build()
            jobScheduler.schedule(info)
        }

        val jobPluginloaderInfo = JobInfo.Builder(Constant.Jobs.JOB_INDEX_INSTALLED_PLUGINS, ComponentName(context, PluginLoader::class.java))
                .setBackoffCriteria(60000, JobInfo.BACKOFF_POLICY_LINEAR)
                .setOverrideDeadline(0)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            jobPluginloaderInfo
                    .setImportantWhileForeground(true)
                    .setPrefetch(true)
        }
        jobScheduler.schedule(jobPluginloaderInfo.build())
    }
}
